src/include/btree_cmp_inline.h

```diff
@@ -88,10 +88,18 @@ __wt_lex_compare(const WT_ITEM *user_item, const WT_ITEM *tree_item, bool prefix
     /*
      * Use the non-vectorized version for the remaining bytes and for the small key sizes.
      */
-    for (; len > 0; --len, ++userp, ++treep)
+    for (; len > 0; --len, ++userp, ++treep) {
+        /*
+         * When prefix is enabled and we are performing lexicographic comparison on schema formats s
+         * or S, we only want to compare the characters before either of them reach a NULL
+         * character. For format S, a NULL character is always at the end of the string, while
+         * format s NULL characters are set for the remaining unused bytes.
+         */
+        if (prefix && (*userp == '\0' || *treep == '\0'))
+            break;
         if (*userp != *treep)
             return (*userp < *treep ? -1 : 1);
-
+    }
```

Comment:
It is possible that prefix is set, and the underlying format is `u` as well. In that case, NUL characters don't need to be account for, since the length doesn't need to account for the NUL character. This if statement represents the fact if the format is either S or s, we want to make sure we don't read in NUL characters.

Reply:
Had a discussion with @etienneptl  and @luke-pearson. We found that it would be useful to limit prefix enabled search near() calls to only formats S, s or u.